enhancements/node-tuning/secure-shortname-image-pulls.md

```diff
@@ -0,0 +1,216 @@
+corresponding fully qualified image names. We can use this new feature to ensure that any Red Hat and OpenShift
+images that are referred to by short name will always resolve to our official locations, hence
+guaranteeing that those images will be official.
+
+### Goals
+
+- Secure where the official Red Hat and OpenShift images are pulled from when using shortnames
+- The alias table list will be maintained by us and be specific to the OpenShift use-case
+- The MCO will be used to lay down the alias table
+
+### Non-Goals
+
+- This is not encouraging the use of shortnames. It is instead focused on making the few use-cases
+  that are out there more secure
+
+## Proposal
+
+Enable the shortname alias logic in the cri-o code so that when an image is being pulled and
+referred to by shortname, cri-o will check the alias table first and if there is a match it
+will pull the image from that location. If there is no match it will fallback to check the
+`unqualified-search-registries` list in order and pull the image from the first location where
+a match is found - note, this is what is done currently.
+
+We will maintain the alias table shipped on the OpenShift node via the MCO. We will create an alias
+table for the official Red Hat and OCP images based on a subset of the aliases in the existing tables
+that are shipped in RHEL. We are doing this as the tables shipped with RHEL have a lot of images from
+the community which are not required in the OpenShift use-case.
+
+Note: This proposal is a security enhancement around the shortnames use case and is not an API change.
+
+### User Stories
+
+#### As a user, I would like to use image short names when running my workloads
+
+When the user uses shortname for official Red Hat and OpenShift images that have a matching alias in the
+alias table, the image will be pulled from the matching location specified there. This will ensure
+the security of where these images are being pulled from reducing the risk of spoofing.
+
+### API Extensions
+
+No API changes are needed for this proposal.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Implementing this enhancement requires changes in:
+
+- cri-o/cri-o
+- openshift/machine-config-operator
+
+Update the cri-o code to enable the shortname alias logic to `permissive` mode. This will ensure
+that when an image is referred to by shortname, it will check the alias table first, if there is no
+match, it will then try all the registries in the `unqualified-search-registries` list in order.
+
+Documentation: We will document that while we highly discourage the use of shortnames, there is a way to have
+more secure shortname access for official Red Hat and OpenShift images.
+
+### Risks and Mitigations
+
+There are no risks involved here as the alias table will be maintained by us in the MCO repo and will only contain aliases for official
+Red Hat and OpenShift images.
```

Comment:
There is a known customer broken by us shipping an alias, in https://github.com/containers/image/issues/1448 .

I’m not at all saying that that customer’s setup is a good reason to not ship aliases, I just think we should, at least, acknowledge that changing the aliases _can_ break existing deployments - both _adding_ aliases and _removing_ aliases. And maybe think about what upgrade processes / guarantees we should make (e.g. “we will never drop an alias”, “we will only ever add aliases that match `openshift*`” — I’m not at all suggesting _these_ policies, just that we might need _some_).

Reply:
This is a good point. I was assuming that we would keep this list as stable as possible with minimal changes to the aliases already added to the list. Will need to have a discussion on how often we expect to update the alias table and plan to handle upgrades/downgrades with that.